src/h5web/visualizations/heatmap/HeatmapVis.tsx

```diff
@@ -21,29 +24,42 @@ function HeatmapVis(props: Props): JSX.Element {
   const values = dataArray.data as number[];
 
   const {
-    dataDomain,
     customDomain,
     scaleType,
     colorMap,
     keepAspectRatio,
     showGrid,
-    initDataDomain,
+    resetDomains,
   } = useHeatmapConfig();
 
-  // width / height <=> cols / rows
-  const aspectRatio = keepAspectRatio ? cols / rows : undefined;
+  const dataDomain = useMemo(() => findDomain(values), [values]);
+  const prevDataDomain = usePrevious(dataDomain);
+
+  // If `dataDomain` just changed, use `undefined` custom domain for this render until config is updated through `useEffect` below
+  const freshCustomDomain =
+    dataDomain !== prevDataDomain ? undefined : customDomain;
```

Comment:
The second part of the solution: we disregard the custom domain from the config if it is potentially stale.

Reply:
I feel we can find a better name than `freshCustomDomain`... But I cannot :sweat_smile: 

Perhaps it is `customDomain` that should be renamed to `sliderDomain` or `toolbarDomain` :shrug: ?

Reply:
I had `effectiveCustomDomain` and `safeCustomDomain` at some point...

Reply:
Not a fan of chaning the name in the config, since it's only during one render that we skip using it.

Reply:
Thinking about it, I stand my point: I think it would make sense to rename `customDomain` to `sliderDomain` or something similar and `freshCustomDomain` to `customDomain`.

Then, the ternary would convey the meaning: 
*If the data domain is updated, set the custom domain to `undefined` else use the customDomain from the store*

Reply:
I could rename it in the destructuring expression, but then if we pass the config as prop... Not sure... :D

Reply:
Yeah okay (sorry GitHub had not updated)